user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/SampleContentComponent.java

```diff
@@ -85,6 +102,56 @@ private void reloadOnBatchRegistration() {
         event -> displayComponentInContent(sampleDetailsComponent));
   }
 
+  private void addBatchUpdateListeners() {
+    batchDetailsComponent.addBatchCreationListener(this::registerBatch);
+    batchDetailsComponent.addBatchDeletionListener(this::openBatchDeletionConfirmation);
+    batchDetailsComponent.addBatchEditListener(this::editBatch);
+    batchDetailsComponent.addBatchViewListener(this::viewBatch);
+  }
+
+  private void registerBatch(CreateBatchEvent createBatchEvent) {
+    onRegisterButtonClicked();
+  }
+
+  //Todo Determine how samples should be loaded and shown within batch
+  private void viewBatch(ViewBatchEvent viewBatchEvent) {
+    ConfirmDialog confirmDialog = new ConfirmDialog();
+    confirmDialog.setText(String.format("This is where I'd show all of my %s Samples",
+        viewBatchEvent.batchPreview().sampleCount()));
+    confirmDialog.open();
+    confirmDialog.addConfirmListener(event -> confirmDialog.close());
+  }
+
+  private void editBatch(EditBatchEvent editBatchEvent) {
+    //Todo Delete Batch in openBis and Database
+  }
+
+  private void deleteBatch(DeleteBatchEvent deleteBatchEvent) {
+    //Todo Delete Batch in openBis and Database
+  }
+
+
+  private void openBatchDeletionConfirmation(DeleteBatchEvent deleteBatchEvent) {
+    BatchDeletionConfirmationNotification batchDeletionConfirmationNotification = new BatchDeletionConfirmationNotification(
+        deleteBatchEvent.batchPreview().sampleCount());
+    batchDeletionConfirmationNotification.open();
+    batchDeletionConfirmationNotification.addConfirmListener(event -> {
+      deleteBatch(deleteBatchEvent);
+      batchDeletionConfirmationNotification.close();
+    });
+    batchDeletionConfirmationNotification.addCancelListener(
+        event -> batchDeletionConfirmationNotification.close());
+  }
+
+  private void onRegisterButtonClicked() {
+    //ToDo Move Batch Registration from SampleDetailsComponent to here
+  }
+
+  //ToDo replace with actual reload
```

Comment:
What is this TODO about?

Reply:
The components within the main component have to be reloaded since we don't propagate batch updates to each component for now. However i removed this since it's only relevant in the PR in which batch updates are tackled